MH-13099, Warn when default credentials are being used

lkiesow opened the pull request:
Using default credentials in production is bad and users should never do
that but we have still seen that happening.

This patch will cause Opencast to print a big warning if the passwords
are not changed. Everything will run as usual (important for testing)
but it's now more likely that users will notice they have a problem.

lkiesow committed:
MH-13099, Warn when default credentials are being used

Using default credentials in production is bad and users should never do
that but we have still seen that happening.

This patch will cause Opencast to print a big warning if the passwords
are not changed. Everything will run as usual (important for testing)
but it's now more likely that users will notice they have a problem.

gregorydlogan commented on modules/userdirectory/src/main/java/org/opencastproject/userdirectory/InMemoryUserAndRoleProvider.java:
```diff
@@ -99,11 +99,26 @@
-    if (digestUsername == null)
+    if (digestUserPass == null) {
       logger.warn("Digest password has not been configured ({})", DIGEST_PASSWORD_KEY);
+    } else if ("CHANGE_ME".equals(digestUserPass)) {
```
Can we make this a constant?  It's lame, but IMO better form.

lkiesow replied:
Done.

gregorydlogan commented on modules/userdirectory/src/main/java/org/opencastproject/userdirectory/AdminUserAndGroupLoader.java:
```diff
@@ -124,6 +124,20 @@ public void activate(ComponentContext cc) throws Exception {
     adminEmail = StringUtils.trimToNull(bundleCtx.getProperty(OPT_ADMIN_EMAIL));
     adminRoles = StringUtils.trimToNull(bundleCtx.getProperty(OPT_ADMIN_ROLES));
 
+    if ("opencast".equals(adminPassword)) {
```
Likewise, constant?

lkiesow committed:
MH-13099, Add constants for the default password configuration

This adds constants to the Java code which represent the default
passwords set in custom.properties for the admin and system user.
